core/network/impl/peer_manager_impl.cpp

```diff
@@ -168,15 +166,27 @@ namespace kagome::network {
 
     // Check if disconnected
     auto block_announce_protocol = router_->getBlockAnnounceProtocol();
+    boost::optional<PeerId> disconnected_peer;
     for (auto it = active_peers_.begin(); it != active_peers_.end();) {
       auto [peer_id, data] = *it++;
       // TODO(d.khaustov) consider better alive check logic
-      if (not stream_engine_->isAlive(peer_id, block_announce_protocol)
-          && clock_->now() - data.time > 5min) {
+      if (not stream_engine_->isAlive(peer_id, block_announce_protocol)) {
         // Found disconnected
-        auto &peer_id_ref = peer_id;
+        const auto &peer_id_ref = peer_id;
         SL_DEBUG(log_, "Found dead peer_id={}", peer_id_ref.toBase58());
         disconnectFromPeer(peer_id);
+        if (boost::none == disconnected_peer) {
```

Comment:
It  is more readable:
```suggestion
        if (not disconnected_peer.has_value()) {
```